### PR TITLE
Move files where possible instead of deleting them.

### PR DESCRIPTION
**Pull request by reitermarkus**
Add helper function for (re)moving files depending on whether the runner is ephemeral or not.

### CODE REVIEW

File: lib/test_cleanup.rb [hunk 1]
```diff
@@ -38,6 +38,24 @@ def reset_if_needed(repository)
       test git, "-C", repository, "reset", "--hard", default_ref
     end
 
+    # Moving files is faster than removing them,
+    # so move them if the current runner is ephemeral.
+    def delete_or_move(paths)
+      paths = Array(paths)
+
+      return if paths.empty?
```
Comment (MikeMcQuaid) [inline lib/test_cleanup.rb:46]:
```suggestion
    def delete_or_move(paths)
      return if paths.blank?
```

File: lib/tests/cleanup_before.rb [hunk 2]
```diff
@@ -25,11 +25,7 @@ def run!(args:)
             # brew doctor complains
             test "sudo", "rm", "-rf", "/usr/local/include/node/"
           elsif OS.mac?
-            if Dir.glob("#{HOMEBREW_CELLAR}/*").present?
-              FileUtils.mkdir_p "/tmp/test-bot-cleanup"
-              # moving is much faster than deleting.
-              test "bash", "-c", "mv #{HOMEBREW_CELLAR}/* /tmp/test-bot-cleanup"
-            end
+            delete_or_move Pathname.glob(HOMEBREW_CELLAR/"*")
```
Comment (MikeMcQuaid) [inline lib/tests/cleanup_before.rb:28]: Could remove the `if ENV["HOMEBREW_GITHUB_ACTIONS"] && !ENV["GITHUB_ACTIONS_HOMEBREW_SELF_HOSTED"]` check on line 22 given `delete_or_move` already handles this?

**Review by MikeMcQuaid — COMMENTED** [review]
Nice 👏🏻

File: lib/tests/cleanup_before.rb (hunk 2 above)
Comment (reitermarkus) [reply lib/tests/cleanup_before.rb:28]: No, because the Linux case doesn't use `delete_or_move`.

Comment (MikeMcQuaid) [reply lib/tests/cleanup_before.rb:28]: @reitermarkus Might be worth testing if it could?

Comment (reitermarkus) [reply lib/tests/cleanup_before.rb:28]: Linux needs `sudo`.

**Comment by fxcoudert** [comment]
Hi @reitermarkus I'm trying to figure out what caused this issue: https://github.com/Homebrew/homebrew-core/issues/66328
(dispatch-build-bottle workflow does not clean the node its runs on). There have been relatively few changes to the testbot recently, so could you please have a look at the issue and tell me if you think this could be the cause? Thanks

**Pull request merged**